src/frontend/utils.ts

```diff
@@ -21,59 +21,71 @@ export default class FrontendUtilities extends Utilities {
   }
 
   /**
-  * Sends a request to real BTT built in webserver with given data translated as GET query params
-  */
+   * Sends a request to real BTT built in webserver with given data translated as GET query params
+   */
   public async callBetterTouchTool(
-    action: string, 
+    action: string,
     data: Types.BttPayload,
     config: Types.AppConfig,
-    translate: boolean = true,
+    translate: boolean = true
   ): Promise<Types.CallResult> {
     if (this.isInBttWebView) {
+      if (config.sharedSecret) {
+        data["shared_secret"] = config.sharedSecret;
+      }
       return this.callBttWebViewFunctions(action, data, translate);
     }
     return this.callWebserverApi(action, data, config, translate);
   }
 
-  public performanceNow = () => { return window.performance.now(); };
+  public performanceNow = () => {
+    return window.performance.now();
+  };
 
   private get isInBttWebView(): boolean {
     const WVWindow: Types.WebViewWindow = window as Types.WebViewWindow;
-    return (Boolean(WVWindow.BTT) && typeof WVWindow.BTT.callHandler === 'function');
+    return (
+      Boolean(WVWindow.BTT) && typeof WVWindow.BTT.callHandler === "function"
+    );
   }
 
   /**
    * Available only on frontend, because this is valid only for built in BTT webview
-   * @param action 
-   * @param data 
-   * @param translate 
+   * @param action
+   * @param data
+   * @param translate
    */
   private callBttWebViewFunctions(
     action: string,
     data: Types.BttPayload,
-    translate: boolean,
+    translate: boolean
```

Comment:
if you end up setting up the linter, please enforce the use of trailing comma :)